middle_end/flambda/inlining/inlining_report.ml

```diff
@@ -0,0 +1,176 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                                 OCaml                                  *)
+(*                                                                        *)
+(*                        Guillaume Bury, OCamlPro                        *)
+(*                                                                        *)
+(*   Copyright 2020--2020 OCamlPro SAS                                    *)
+(*                                                                        *)
+(*   All rights reserved.  This file is distributed under the terms of    *)
+(*   the GNU Lesser General Public License version 2.1, with the          *)
+(*   special exception on linking described in the file LICENSE.          *)
+(*                                                                        *)
+(**************************************************************************)
+
+(* Decisions related to inlining,
+
+   This includes deciding not to consider some function for inlining
+   (e.g. method, c-calls, or function with no precise enough type to
+   inline). *)
+
+module Decision = struct
+
+  module At_call_site = struct
+
+    type t =
+      | C_call
+      | Method
+      | Unknown_function
+      | Non_inlinable_function of {
+          code_id : Code_id.t;
+        }
+      | Inlinable_function of {
+          code_id : Code_id.t;
+          decision : Inlining_decision.Call_site_decision.t;
+        }
+
+  end
+
+  module At_function_declaration = struct
+
+    type pass =
+      | Before_simplify
+      | After_simplify
+
+    type t = {
+      pass : pass;
+      code_id : Code_id.t;
+      decision : Inlining_decision.Function_declaration_decision.t;
+    }
+
+  end
+
+  type t =
+    | At_call_site of At_call_site.t
+    | At_function_declaration of At_function_declaration.t
+
+end
```

Comment:
Same comment as for `inlining_cost.ml`: I don't really like the extra module structure when you only have types defined.